### PR TITLE
feat: add custom header support

### PR DESCRIPTION
**Pull request by eshanholtz**

### CODE REVIEW

File: src/Twilio/Http/Request.cs [hunk 1]
```diff
@@ -238,8 +258,12 @@ public override bool Equals(object obj)
...
             return Method.Equals(other.Method) &&
                    buildUri().Equals(other.buildUri()) &&
-                   QueryParams.All(other.QueryParams.Contains) &&
-                   PostParams.All(other.PostParams.Contains);
+                   QueryParams.All(q => other.QueryParams.Contains(q)) &&
+                   other.QueryParams.All(qOther => QueryParams.Contains(qOther)) &&
+                   PostParams.All(p => other.PostParams.Contains(p)) &&
+                   other.PostParams.All(pOther => PostParams.Contains(pOther)) &&
+                   HeaderParams.All(h => other.HeaderParams.Contains(h)) &&
+                   other.HeaderParams.All(hOther => HeaderParams.Contains(hOther));
```
Comment (childish-sambino) [inline src/Twilio/Http/Request.cs:266]: Oh, nice catch. But can the old syntax still be used?

File: src/Twilio/Rest/Accounts/V1/Credential/PublicKeyResource.cs [hunk 2]
```diff
@@ -154,7 +155,8 @@ private static Request BuildCreateRequest(CreatePublicKeyOptions options, ITwili
                 HttpMethod.Post,
                 Rest.Domain.Accounts,
                 "/v1/Credentials/PublicKeys",
-                postParams: options.GetParams()
+                postParams: options.GetParams(),
+                headerParams: null
```
Comment (childish-sambino) [inline src/Twilio/Rest/Accounts/V1/Credential/PublicKeyResource.cs:159]: Can these be reverted?

Comment (eshanholtz) [reply src/Twilio/Rest/Accounts/V1/Credential/PublicKeyResource.cs:159]: Absolutely. Just need to add some additional logic

File: src/Twilio/Http/Request.cs (hunk 1 above)
Comment (eshanholtz) [reply src/Twilio/Http/Request.cs:266]: Not sure, will test and update if it can

File: src/Twilio/Http/Request.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
```
Comment (childish-sambino) [inline src/Twilio/Http/Request.cs:3]: Is this needed?

**Review by childish-sambino — APPROVED** [review]

File: src/Twilio/Http/Request.cs (hunk 3 above)
Comment (eshanholtz) [reply src/Twilio/Http/Request.cs:3]: Ugh darn, remnant from local testing. nice catch, thanks.

**Pull request merged**